### PR TITLE
feat: add ingredients in new recipe form

### PR DESCRIPTION
**Pull request by lonba**

### CODE REVIEW

File: components/recipeForms/newRecipeForm.tsx [hunk 1]
```diff
@@ -1,10 +1,16 @@
-import { useForm, SubmitHandler } from 'react-hook-form'
+import React, { MouseEvent, useEffect } from 'react';
+import { useForm, SubmitHandler, useFieldArray } from 'react-hook-form'
 import { useMutation } from 'urql'
 
 interface IFormInput {
```
Comment (moncapitaine) [inline components/recipeForms/newRecipeForm.tsx:5]: why not using Recipe but re-invent a new interface?

File: components/recipeForms/newRecipeForm.tsx [hunk 2]
```diff
@@ -15,25 +21,55 @@ interface Dish {
...
+    const areAnyIngredientsListed = (dishIngredients.length > 0)
+
+    if(!areAnyIngredientsListed) {
+      appendIngredient({ amount: "", name: ""})
+    }
+  }, [])
+
   return (
     <div className="container">
       <a href="#" className="text-1 font-bold text-zinc-400 hover:text-black">
```
Comment (moncapitaine) [inline components/recipeForms/newRecipeForm.tsx:75]: use the router and it's back() method

File: components/recipeForms/newRecipeForm.tsx (hunk 1 above)
Comment (lonba) [reply components/recipeForms/newRecipeForm.tsx:5]: fixed with import Recipe from '../../interfaces/recipe';

File: components/recipeForms/newRecipeForm.tsx [hunk 3]
```diff
@@ -15,25 +21,55 @@ interface Dish {
...
+  }  
+  const removeIngredientRowHandler = (e: MouseEvent<HTMLButtonElement>, index: number) => {
+    e.preventDefault()
+    console.log(`removeIngredientRowHandler with index ${index}`)
+    removeIngredient(index)
   }
 
+  useEffect(() => {
+    const { dishIngredients } = getValues()
+    const areAnyIngredientsListed = (dishIngredients.length > 0)
```
[the remark this answers is not in the document]
Comment (lonba) [reply components/recipeForms/newRecipeForm.tsx:66]: solved

File: components/recipeForms/newRecipeForm.tsx [hunk 4]
```diff
@@ -60,18 +96,47 @@ const NewRecipeForm = () => {
...
-            className="border-2 rounded-lg w-1/4 py-1 px-3"
-            {...register('dishIngredients')}
-          />
-          <input
-            className="border-2 rounded-lg w-3/4 py-1 px-3"
-            {...register('dishIngredients')}
-          />
-        </div>
+
+        {ingredientFields.map( (ingredient, index, field) => {
```
[the remark this answers is not in the document]
Comment (lonba) [reply components/recipeForms/newRecipeForm.tsx]: moved into component AddRecipeIngredient

File: components/recipeForms/newRecipeForm.tsx (hunk 2 above)
Comment (lonba) [reply components/recipeForms/newRecipeForm.tsx]: merge from main resolved the issue

File: components/recipeForms/newRecipeForm.tsx [hunk 5]
```diff
@@ -1,42 +1,38 @@
...
+    createRecipe(input: $input) {
+      id
+      name
+      description
+    }
+  }`)
+  
+  const onSubmit: SubmitHandler<Recipe> = async (data: Recipe) => {
+    /*
+    //prevent spamming to backend
```
Comment (moncapitaine) [inline components/recipeForms/newRecipeForm.tsx:23]: remove comment

**Review by moncapitaine — APPROVED** [review]

**Pull request merged**